src/indieweb_utils/indieauth/profile.py

```diff
@@ -1,72 +1,79 @@
 from dataclasses import dataclass
+from typing import Optional
 
 import requests
-
 from bs4 import BeautifulSoup
 
+from . import constants
+
 
 class ProfileError(Exception):
     pass
 
 
-@dataclass
+@dataclass(frozen=True)
 class Profile:
     name: str
-    photo: str
+    photo: Optional[str]
     url: str
-    email: str
+    email: Optional[str]
 
 
-def get_profile(
-        me: str
-    ) -> Profile:
+def get_profile(me: str) -> Profile:
 
     try:
-        me_profile = requests.get(me)
-    except:
+        me_profile = requests.get(me, timeout=10)
+    except requests.exceptions.RequestException:
         raise ProfileError("Request to retrieve profile URL did not return a valid response.")
 
     profile_item = BeautifulSoup(me_profile.text, "html.parser")
-    h_card = profile_item.select(".h-card")
-
-    if not h_card:
+    h_card_tag = profile_item.select(".h-card")
+    try:
+        h_card = h_card_tag[0]
+    except TypeError:
+        return Profile(name="", photo=None, url="", email=None)
```

Comment:
Should name and url both be `me` since it acts as a default value when we do have an h-card but can't find the value?

Reply:
I support that behaviour.